stylelint-polaris/plugins/custom-properties-allowed-list.js

```diff
@@ -209,8 +210,7 @@ function validateAllowedPropertiesOption(option, invalidOptions) {
  * @param {InvalidOptions} invalidOptions
  */
 function validateAllowedValuesOption(option, invalidOptions) {
-  /** @type {PrimaryOptions['allowedValues']} */
-  const allowedValues = option;
+  const allowedValues = /** @type {PrimaryOptions['allowedValues']} */ (option);
```

Comment:
Perhaps `@param {unknown} option` should be `@param {PrimaryOptions['allowedValues']}` then you wouldn't need this cast?

Reply:
I think we went a little overboard with the options validation in the initial implementation 😅  I removed the extraneous validation logic in favor of a simple error message informing the user of the malformed option configuration: https://github.com/Shopify/polaris/pull/5331/commits/0dec735769bcb3de6135ea4a3da50f5ccd03dbda

Resolving as the change marks this conversation `outdated`. Feel free to unresolve or start a new discussion if you have feedback on the updated implementation.